--- ../vendor/k8s.io/kubernetes/pkg/scheduler/framework/preemption/preemption.go	2026-01-26 00:00:00
+++ ../vendor/k8s.io/kubernetes/pkg/scheduler/framework/preemption/preemption.go	2026-01-26 00:00:00
@@ -20,7 +20,9 @@
 	"context"
 	"errors"
 	"fmt"
 	"math"
+	"os"
+	"strconv"
 	"sync"
 
 	v1 "k8s.io/api/core/v1"
@@ -76,10 +78,19 @@
 func NewEvaluator(pluginName string, fh fwk.Handle, i Interface, executor *Executor) *Evaluator {
+	// FIX kubernetes 1.21 and lower version API missing issue
+	var pdbLister policylisters.PodDisruptionBudgetLister
+	minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+	if minorVersionStr != "" {
+		minorVersion, err := strconv.Atoi(minorVersionStr)
+		if err == nil && minorVersion >= 21 {
+			pdbLister = fh.SharedInformerFactory().Policy().V1().PodDisruptionBudgets().Lister()
+		}
+	}
 	return &Evaluator{
 		PluginName: pluginName,
 		Handler:    fh,
 		PodLister:  fh.SharedInformerFactory().Core().V1().Pods().Lister(),
-		PdbLister:  fh.SharedInformerFactory().Policy().V1().PodDisruptionBudgets().Lister(),
+		PdbLister:  pdbLister,
 		Interface:  i,
 		executor:   executor,
 	}
 }
