--- ../vendor/k8s.io/kubernetes/pkg/scheduler/eventhandlers.go
+++ ../vendor/k8s.io/kubernetes/pkg/scheduler/eventhandlers.go
@@ -19,6 +19,8 @@
 import (
 	"context"
 	"fmt"
+	"os"
+	"strconv"
 	"strings"
 	"time"
 
@@ -579,6 +581,14 @@
 			}
 			handlers = append(handlers, handlerRegistration)
 		case fwk.CSIStorageCapacity:
+			// FIX kubernetes 1.23 and lower version API missing issue
+			minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+			if minorVersionStr != "" {
+				minorVersion, err := strconv.Atoi(minorVersionStr)
+				if err != nil || minorVersion < 24 {
+					continue
+				}
+			}
 			if handlerRegistration, err = informerFactory.Storage().V1().CSIStorageCapacities().Informer().AddEventHandler(
 				buildEvtResHandler(at, fwk.CSIStorageCapacity),
 			); err != nil {
@@ -614,6 +624,14 @@
 			}
 			handlers = append(handlers, handlerRegistration)
 		case fwk.ResourceClaim:
+			// FIX kubernetes lower version API missing issue
+			minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+			if minorVersionStr != "" {
+				minorVersion, err := strconv.Atoi(minorVersionStr)
+				if err != nil || minorVersion < 34 {
+					continue
+				}
+			}
 			if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
 				handlerRegistration = resourceClaimCache.AddEventHandler(
 					buildEvtResHandler(at, fwk.ResourceClaim),
@@ -621,6 +639,14 @@
 				handlers = append(handlers, handlerRegistration)
 			}
 		case fwk.ResourceSlice:
+			// FIX kubernetes lower version API missing issue
+			minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+			if minorVersionStr != "" {
+				minorVersion, err := strconv.Atoi(minorVersionStr)
+				if err != nil || minorVersion < 34 {
+					continue
+				}
+			}
 			if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
 				if handlerRegistration, err = resourceSliceTracker.AddEventHandler(
 					buildEvtResHandler(at, fwk.ResourceSlice),
@@ -630,6 +656,14 @@
 				handlers = append(handlers, handlerRegistration)
 			}
 		case fwk.DeviceClass:
+			// FIX kubernetes lower version API missing issue
+			minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+			if minorVersionStr != "" {
+				minorVersion, err := strconv.Atoi(minorVersionStr)
+				if err != nil || minorVersion < 34 {
+					continue
+				}
+			}
 			if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
 				handler := cache.ResourceEventHandler(buildEvtResHandler(at, fwk.DeviceClass))
 				if utilfeature.DefaultFeatureGate.Enabled(features.DRAExtendedResource) {
