--- ../vendor/k8s.io/kubernetes/pkg/scheduler/eventhandlers.go	2026-01-26 00:00:00
+++ ../vendor/k8s.io/kubernetes/pkg/scheduler/eventhandlers.go	2026-01-26 00:00:00
@@ -19,4 +19,6 @@
 	"context"
 	"fmt"
+	"os"
+	"strconv"
 	"strings"
 	"time"
@@ -600,3 +602,11 @@
 		case fwk.CSIStorageCapacity:
+			// FIX kubernetes 1.23 and lower version API missing issue
+			minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+			if minorVersionStr != "" {
+				minorVersion, err := strconv.Atoi(minorVersionStr)
+				if err != nil || minorVersion < 24 {
+					continue
+				}
+			}
 			if handlerRegistration, err = informerFactory.Storage().V1().CSIStorageCapacities().Informer().AddEventHandler(
 				buildEvtResHandler(at, fwk.CSIStorageCapacity),
@@ -635,3 +643,11 @@
 		case fwk.ResourceClaim:
+			// FIX kubernetes lower version API missing issue
+			minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+			if minorVersionStr != "" {
+				minorVersion, err := strconv.Atoi(minorVersionStr)
+				if err != nil || minorVersion < 34 {
+					continue
+				}
+			}
 			if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
 				handlerRegistration = resourceClaimCache.AddEventHandler(
@@ -642,3 +658,11 @@
 		case fwk.ResourceSlice:
+			// FIX kubernetes lower version API missing issue
+			minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+			if minorVersionStr != "" {
+				minorVersion, err := strconv.Atoi(minorVersionStr)
+				if err != nil || minorVersion < 34 {
+					continue
+				}
+			}
 			if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
 				if handlerRegistration, err = resourceSliceTracker.AddEventHandler(
@@ -651,3 +675,11 @@
 		case fwk.DeviceClass:
+			// FIX kubernetes lower version API missing issue
+			minorVersionStr := os.Getenv("KUBE_API_VERSION_MINOR")
+			if minorVersionStr != "" {
+				minorVersion, err := strconv.Atoi(minorVersionStr)
+				if err != nil || minorVersion < 34 {
+					continue
+				}
+			}
 			if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
 				handler := cache.ResourceEventHandler(buildEvtResHandler(at, fwk.DeviceClass))
