--- ../vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumebinding/binder.go	2026-01-26 00:00:00
+++ ../vendor/k8s.io/kubernetes/pkg/scheduler/framework/plugins/volumebinding/binder.go	2026-01-26 00:00:00
@@ -276,4 +276,7 @@
 	b.csiDriverLister = capacityCheck.CSIDriverInformer.Lister()
-	b.csiStorageCapacityLister = capacityCheck.CSIStorageCapacityInformer.Lister()
+	// FIX kubernetes 1.24 and lower version API missing issue
+	if capacityCheck.CSIStorageCapacityInformer != nil {
+		b.csiStorageCapacityLister = capacityCheck.CSIStorageCapacityInformer.Lister()
+	}
 
 	return b, nil
@@ -999,5 +999,9 @@
 	if driver.Spec.StorageCapacity == nil || !*driver.Spec.StorageCapacity {
 		return true, nil, nil
 	}
+	// FIX kubernetes 1.24 and lower version API missing issue
+	if b.csiStorageCapacityLister == nil {
+		return true, nil, nil
+	}
 
 	// Look for a matching CSIStorageCapacity object(s).
